crates/ide-assists/src/handlers/replace_if_let_with_match.rs

```diff
@@ -728,6 +728,363 @@ fn foo(x: Result<i32, ()>) {
         );
     }
 
+    #[test]
+    fn test_if_let_with_match_variant_nested_or_literal() {
```

Comment:
I think this test can benefit from a split to some tests, but I'll leave that to you.